### PR TITLE
Optimize assign

### PR DESCRIPTION
**Pull request by jacquesqiao**

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/test_layers.py [hunk 1]
```diff
@@ -1015,6 +1015,18 @@ def test_batch_norm(self):
...
+        import numpy as np
+        startup = Program()
+        main = Program()
+        with program_guard(main, startup):
+            probs = layers.assign(
+                input=np.random.random([1, 2]).astype('float32'),
+                init_once=True)
+
+        print(str(main))
+        print(str(startup))
```
Comment (panyx0718) [inline python/paddle/fluid/tests/unittests/test_layers.py]: this is a bad test...

File: python/paddle/fluid/layers/nn.py [hunk 2]
```diff
@@ -5010,10 +5010,12 @@ def nce(input,
             alias_probs_[little[0]] = 1.0
             alias_[little[0]] = -1
 
-        probs = assign(input=np.array(custom_dist).astype('float32'))
-        custom_alias = assign(input=np.array(alias_).astype('int32'))
+        probs = assign(
+            input=np.array(custom_dist).astype('float32'), init_once=True)
```
Comment (panyx0718) [inline python/paddle/fluid/layers/nn.py]: why not create var in startup program here?

Comment (typhoonzero) [reply python/paddle/fluid/layers/nn.py]: agree to create var here.

File: python/paddle/fluid/tests/unittests/test_layers.py (hunk 1 above)
Comment (jacquesqiao) [reply python/paddle/fluid/tests/unittests/test_layers.py]: real test is here https://github.com/PaddlePaddle/Paddle/pull/15080/files#diff-68cdbd1cba20d9e8a5ddfcdd1ffa88f6L35. It's disabled now, i enabled it and use it for the test.

File: python/paddle/fluid/layers/nn.py (hunk 2 above)
Comment (jacquesqiao) [reply python/paddle/fluid/layers/nn.py]: persistable var must be created both in startup program and main program for type and shape inference.

Comment (typhoonzero) [reply python/paddle/fluid/layers/nn.py]: `assign` should be doing what the name is. or you can add a layer named `layers.static_tensor` but not an attirbute on `assign`

Comment (jacquesqiao) [reply python/paddle/fluid/layers/nn.py]: fix done

File: python/paddle/fluid/initializer.py [hunk 3]
```diff
@@ -683,6 +684,64 @@ def __call__(self, var, block):
...
+        if dtype == VarDesc.VarType.FP32:
+            value_name = "fp32_values"
+            values = [float(v) for v in self._value.flat]
+        elif dtype == VarDesc.VarType.INT32:
+            value_name = "int32_values"
+            values = [int(v) for v in self._value.flat]
+        else:
+            raise ValueError("Unsupported dtype %s", self._value.dtype)
+        if self._value.size > 1024 * 1024 * 5:
+            raise ValueError("The size of input is too big. Please consider "
```
Comment (typhoonzero) [inline python/paddle/fluid/initializer.py:730]: warn the max size

**Review by typhoonzero — APPROVED** [review]

**Review by panyx0718 — APPROVED** [review]
LG API

**Pull request merged**